libs/rand/random.go

```diff
@@ -54,96 +42,115 @@ func (r *Rand) reset(seed int64) {
 // Global functions
 
 func Seed(seed int64) {
```

Comment:
I think the idea in the original issue was to get rid of these global functions completely. Instead, each "thread of execution" (i.e. caller) would have a `rand.New()` instance (probably as part of the main struct) and call `rand.XXX` (e.g. rand.Intn) each time it needs a random unit

Reply:
ok, got it.